Sources/MongoSwift/Operations/RunCommandOperation.swift

```diff
@@ -42,10 +42,22 @@ internal struct RunCommandOperation: Operation {
     internal func execute() throws -> Document {
         let rp = self.options?.readPreference ?? self.database.readPreference
         let opts = try encodeOptions(options: self.options, session: self.session)
-        let reply = Document()
+        var reply = Document()
         var error = bson_error_t()
-        guard mongoc_database_command_with_opts(
-            self.database._database, self.command.data, rp?._readPreference, opts?.data, reply.data, &error) else {
+
+        let success =
+        self.command.withBSONPointer { cmdPtr in
+            withOptionalBSONPointer(to: opts) { optsPtr in
+                reply.withMutableBSONPointer { replyPtr in
+                    withUnsafeMutablePointer(to: &error) { errPtr in
+                        mongoc_database_command_with_opts(
+                            self.database._database, cmdPtr, rp?._readPreference, optsPtr, replyPtr, errPtr)
+                    }
+                }
+            }
+        }
```

Comment:
while I appreciate the added explicitness of what we're doing, this pyramid like growth for each argument we pass to mongoc is pretty large in cases like this.

What do we think about only using such helpers where where they're necessary (e.g. mutable `bson_t` pointers or where we run into compiler issues) and rely on `&` otherwise? I think a large use-case of these helpers is if you want to coerce the pointer into something else. Since we're just passing them as-is, I imagine the same stuff is going on under the hood as if we just used `&`.

Reply:
yeah, all the nesting is a bit annoying. 
I would be ok with making the `.data` property return a `BSONPointer` and using that for read-only access cases. then this one would become
```swift
let success =
reply.withMutableBSONPointer { replyPtr in
    withUnsafeMutablePointer(to: &error) { errPtr in
        mongoc_database_command_with_opts(
            self.database._database, cmd.data, rp?._readPreference, opts?.data, replyPtr, errPtr)
    }
}
```

Reply:
what do you think @mbroadst ?

Reply:
Discussed offline. We agreed that any class that wraps a pointer must provide a mutable accessor method. Passing local variables by reference is fine (e.g. `&error`)

Reply:
Yeah, I think it looks much better now 👍